src/System/AtomicWrite/Writer/LazyByteString/Binary.hs

```diff
@@ -0,0 +1,53 @@
+-- |
+-- Module      :  System.AtomicWrite.Writer.LazyByteString.Binary
+-- Copyright   :  © 2015-2017 Stack Builders Inc.
+-- License     :  MIT
+--
+-- Maintainer  :  Stack Builders <hackage@stackbuilders.com>
+-- Stability   :  experimental
+-- Portability :  portable
+--
+-- Provides functionality to dump the contents of a Lazy ByteString
+-- to a file in binary mode.
+
+module System.AtomicWrite.Writer.LazyByteString.Binary (atomicWriteFile, atomicWriteFileWithMode) where
+
+import           System.AtomicWrite.Internal (closeAndRename, maybeSetFileMode,
+                                              tempFileFor)
+
+import           Data.ByteString.Lazy        (ByteString, hPutStr)
+
+import           System.Posix.Types          (FileMode)
+
+import           System.IO                   (hSetBinaryMode)
+
+-- | Creates or modifies a file atomically on POSIX-compliant
+-- systems while preserving permissions.
+atomicWriteFile ::
+  FilePath      -- ^ The path where the file will be updated or created
+  -> ByteString -- ^ The content to write to the file
+  -> IO ()
+atomicWriteFile =
+  atomicWriteFileMaybeMode Nothing
+
+-- | Creates or modifies a file atomically on
+-- POSIX-compliant systems and updates permissions
+atomicWriteFileWithMode ::
+  FileMode      -- ^ The mode to set the file to
+  -> FilePath   -- ^ The path where the file will be updated or created
+  -> ByteString -- ^ The content to write to the file
+  -> IO ()
+atomicWriteFileWithMode mode =
+  atomicWriteFileMaybeMode $ Just mode
+
+-- Helper Function
+atomicWriteFileMaybeMode ::
+  Maybe FileMode -- ^ The mode to set the file to
+  -> FilePath    -- ^ The path where the file will be updated or created
+  -> ByteString  -- ^ The content to write to the file
+  -> IO ()
+atomicWriteFileMaybeMode mmode path text =
+  tempFileFor path >>= \(tmpPath, h) -> hSetBinaryMode h True
+                    >> hPutStr h text
```

Comment:
These 3 lines are repeated on many files. Maybe, they can be moved to an internal helper and then just be imported in all the files where it's needed

Reply:
I could do four versions of the functions text/binary and text/action.

Reply:
sounds good

Reply:
Although I think we can't because text is of different type on each file
and the function hPutStr is also of different type. Unless, we add extra parameters to the function something like:
f :: Maybe Mode -> FilePath -> (Handle -> a -> IO) -> a -> IO ()